apis/vmware/v1beta1/vspheremachine_types.go

```diff
@@ -34,7 +35,95 @@ type VSphereMachineVolume struct {
 	StorageClass string `json:"storageClass,omitempty"`
 }
 
+// RouteSpec defines a static route for a guest.
+type RouteSpec struct {
+	// To is either "default", or an IP4 address. IP6 is not supported yet.
+	//
+	// +kubebuilder:validation:Pattern=`^(default|([0-9]{1,3}\.){3}[0-9]{1,3}(\/[0-9]{1,2})?)$`
+	To string `json:"to"`
+
+	// Via is an IP4 address. IP6 is not supported yet.
+	//
+	// +kubebuilder:validation:Pattern=`^([0-9]{1,3}\.){3}[0-9]{1,3}$`
+	Via string `json:"via"`
+}
+
+// PartialObjectRef describes a reference to another object in the same
+// namespace as the referrer. The reference can be just a name but may also
+// include the referred resource's APIVersion and Kind.
+type PartialObjectRef struct {
+	metav1.TypeMeta `json:",inline"`
```

Comment:
Please add Kind and APIVersion directly

Please copy over the validation from: https://github.com/kubernetes-sigs/cluster-api/blob/3adb25101d845af093cc44975b67cc1b94838e2e/api/core/v1beta2/machinehealthcheck_types.go#L160-L183